pilot/pkg/xds/ep_filters.go

```diff
@@ -103,6 +107,13 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocalityEndpoint
 				continue
 			}
 
+			// A gateway is considered healthy only if it has any healthy endpoints reachable.
+			if istioEndpoint.HealthStatus == model.Healthy {
+				for _, gw := range gateways {
```

Comment:
I am concerned about the performance

Reply:
I think in most cases a remote network's gateways are a handful of endpoints, not sure if iterating over a few items here would have a significant impact.

Reply:
Nope, remote network can be common as local network

Reply:
> Nope, remote network can be common as local network

To be honest I don't really understand what you mean by that. I would greatly appreciate if you could elaborate on it and why you think there is a performance concern.

The very next line of code, a call to `splitWeightAmongGateways()`, is doing exactly the same in that function - iterating over the list of gateways to update their aggregate weights in a map. If you're specifically concerned that I duplicate that iteration, it can be refactored and will likely be gone when I fix the patch for the rest of the comments.

Reply:
There are two loops